File: cameravision/scripts/ensure-7zip-stub.js
const fs = require('fs');
const path = require('path');

const projectRoot = path.resolve(__dirname, '..');
const stubSrc = path.join(projectRoot, 'stubs', '7zip-bin');
const dest = path.join(projectRoot, 'node_modules', 'builder-util', 'node_modules', '7zip-bin');

try {
  if (!fs.existsSync(stubSrc)) {
    console.error('Stub source not found:', stubSrc);
    process.exit(1);
  }
  if (fs.existsSync(dest)) {
    console.log('7zip-bin stub already present at', dest);
    process.exit(0);
  }
  fs.mkdirSync(path.dirname(dest), { recursive: true });
  if (fs.cpSync) {
    fs.cpSync(stubSrc, dest, { recursive: true });
  } else {
    const copyRecursive = (src, dst) => {
      fs.mkdirSync(dst, { recursive: true });
      for (const entry of fs.readdirSync(src, { withFileTypes: true })) {
        const s = path.join(src, entry.name);
        const d = path.join(dst, entry.name);
        if (entry.isDirectory()) copyRecursive(s, d);
        else fs.copyFileSync(s, d);
      }
    };
    copyRecursive(stubSrc, dest);
  }
  console.log('Copied 7zip-bin stub to', dest);
} catch (err) {
  console.error('Failed to ensure 7zip-bin stub:', err);
  process.exit(1);
}

File: cameravision/scripts/ensure-7zip-stub.py
import os
import shutil
import sys

project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
stub_src = os.path.join(project_root, "stubs", "7zip-bin")
dest = os.path.join(project_root, "node_modules", "builder-util", "node_modules", "7zip-bin")

def fail(msg, code=1):
    print(msg, file=sys.stderr)
    sys.exit(code)

try:
    if not os.path.exists(stub_src):
        fail(f"Stub source not found: {stub_src}")

    if os.path.exists(dest):
        print(f"7zip-bin stub already present at: {dest}")
        sys.exit(0)

    os.makedirs(os.path.dirname(dest), exist_ok=True)
    shutil.copytree(stub_src, dest)
    print(f"Copied 7zip-bin stub to: {dest}")
    sys.exit(0)
except Exception as e:
    fail(f"Failed to ensure 7zip-bin stub: {e}")